presentation/src/main/java/me/androidbox/presentation/login/viewmodel/LoginViewModel.kt

```diff
@@ -4,11 +4,23 @@ import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import me.androidbox.domain.authentication.NetworkResponseState
+import me.androidbox.domain.authentication.model.LoginModel
+import me.androidbox.domain.authentication.usecase.LoginUseCase
 import javax.inject.Inject
 
 @HiltViewModel
-class LoginViewModel @Inject constructor() : ViewModel() {
+class LoginViewModel @Inject constructor(
+    private val loginUseCase: LoginUseCase
+) : ViewModel() {
+
+    private val loginMutableState: MutableStateFlow<NetworkResponseState<LoginModel>?> = MutableStateFlow(null)
+    val loginState = loginMutableState.asStateFlow()
 
     var username by mutableStateOf("")
         private set
```

Comment:
Why do you use Stateflow on one,a nd mutableStateOf on the other?

Also, username isn't a thing in any part of the app

Login uses email/password, and Registration uses email/password/full name

Reply:
I guess I was just experimenting with both. Yes, I should be consistent